src/javascript/app_2/store/pages/statement/statement_model.js

```diff
@@ -0,0 +1,84 @@
+import { observable, action }     from 'mobx';
+import moment                     from 'moment';
+import WS                         from '../../../data/ws_methods';
+import { formatMoney }            from '../../../../_common/base/currency_base';
+import Client                     from '../../../../_common/base/client_base';
+import { localize }               from '../../../../_common/localize';
+import { toTitleCase }            from '../../../../_common/string_util';
+
+const batch_size = 100; // request response limit
+
+export default class LoginHistoryModel {
+    @observable data            = [];
+    @observable is_loading      = false;
+    @observable has_loaded_all  = false;
+    @observable date_from       = '';
+    @observable date_to         = '';
+
+    @action.bound
+    clearTable() {
+        this.data            = [];
+        this.has_loaded_all  = false;
+        this.is_loading      = false;
+    }
+
+    @action.bound
+    fetchNextBatch() {
+        if (this.has_loaded_all || this.is_loading) return;
+
+        this.is_loading = true;
+
+        const currency = Client.get('currency');
+
+        WS.statement(
+            batch_size,
+            this.data.length,
+            {
+                ...this.date_from && {date_from: moment(this.date_from).unix()},
+                ...this.date_to   && {date_to: moment(this.date_to).add(1, 'd').subtract(1, 's').unix()},
+            }
+        ).then((response) => {
+            const formatted_transactions = response.statement.transactions
+                .map(transaction => getStatementData(transaction, currency));
+
+            this.data           = [...this.data, ...formatted_transactions];
+            this.has_loaded_all = formatted_transactions.length < batch_size;
+            this.is_loading     = false;
+        });
+    }
+
+    @action.bound
+    handleDateChange(e) {
+        if (e.target.value !== this[e.target.name]) {
+            this[e.target.name] = e.target.value;
+            this.clearTable();
+            this.fetchNextBatch();
+        }
+    }
+}
+
+/*
+    RESPONSE PARSING FUNCTION
+*/
+const getStatementData = (statement, currency) => {
+    const date_obj   = new Date(statement.transaction_time * 1000);
```

Comment:
@ruskakimov 
There are some helper functions in [`app_2/common/date_time.js`](https://github.com/binary-com/binary-static/blob/new-app/src/javascript/app_2/common/date_time.js) to convert `epoch` to `moment` and `GMT` format.